x-pack/plugins/search_playground/server/routes.ts

```diff
@@ -90,76 +90,82 @@ export function defineRoutes({
-
-        try {
-          sourceFields = JSON.parse(data.source_fields);
-        } catch (e) {
-          log.error('Failed to parse the source fields', e);
-          throw Error(e);
-        }
-
-        const chain = ConversationalChain({
-          model,
-          rag: {
-            index: data.indices,
-            retriever: createRetriever(data.elasticsearch_query),
-            content_field: sourceFields,
-            size: Number(data.doc_size),
-          },
-          prompt: Prompt(data.prompt, {
-            citations: data.citations,
-            context: true,
-            type: 'openai',
-          }),
-        });
-
-        const stream = await chain.stream(aiClient, messages);
+      const aiClient = Assist({
+        es_client: client.asCurrentUser,
+      } as AssistClientOptionsWithClient);
+      const { messages, data } = await request.body;
+      const abortController = new AbortController();
+      const abortSignal = abortController.signal;
+      const model = new ActionsClientChatOpenAI({
+        actions,
+        logger: log,
+        request,
+        connectorId: data.connector_id,
+        model: data.summarization_model,
+        traceId: uuidv4(),
+        signal: abortSignal,
+        // prevents the agent from retrying on failure
+        // failure could be due to bad connector, we should deliver that result to the client asap
+        maxRetries: 0,
+      });
 
-        const { end, push, responseWithHeaders } = streamFactory(request.headers, log);
+      let sourceFields = {};
 
-        const reader = (stream as ReadableStream).getReader();
-        const textDecoder = new TextDecoder();
+      try {
+        sourceFields = JSON.parse(data.source_fields);
+      } catch (e) {
+        log.error('Failed to parse the source fields', e);
+        throw Error(e);
+      }
 
-        async function pushStreamUpdate() {
-          reader.read().then(({ done, value }: { done: boolean; value?: Uint8Array }) => {
-            if (done) {
-              end();
-              return;
-            }
-            push(textDecoder.decode(value));
-            pushStreamUpdate();
-          });
-        }
+      const chain = ConversationalChain({
+        model,
+        rag: {
+          index: data.indices,
+          retriever: createRetriever(data.elasticsearch_query),
+          content_field: sourceFields,
+          size: Number(data.doc_size),
+        },
+        prompt: Prompt(data.prompt, {
+          citations: data.citations,
+          context: true,
+          type: 'openai',
+        }),
+      });
 
-        pushStreamUpdate();
+      let stream: ReadableStream<Uint8Array>;
 
-        return response.ok(responseWithHeaders);
+      try {
+        stream = await chain.stream(aiClient, messages);
       } catch (e) {
         log.error('Failed to create the chat stream', e);
 
-        throw Error(e);
+        return response.badRequest({
+          body: {
+            message: e,
```

Comment:
`message` should be a `string` correct? `e` is `unknown` by default so we should have some handling to parse `e` to a `string` here, right?

Reply:
For OpenAI, they handle the errors from connectors and polish them to be specific error messages instead of `Error` object. Thats why I added the `error message` directly into `message`.

Reply:
But by default `e: unknown` and at runtime it could be anything. we either need a helper function or a guard, something like: 

```
if (typeof e === 'string')
{
  return response.badRequest({
    body: {
      message: e,
    },
  });
} 
throw e;
```

Reply:
updated as suggested